transbank/onepay/transaction.py

```diff
@@ -0,0 +1,122 @@
+# encoding: utf-8
+import calendar
+import http.client
+import json
+
+from datetime import datetime
+from enum import Enum
+from urllib.parse import urlparse
+
+from transbank.onepay.schema import ItemSchema, TransactionCreateRequestSchema, TransactionCreateResponseSchema, SendTransactionResponseSchema
+
+from transbank.onepay.cart import ShoppingCart
+from transbank.onepay.error import TransactionCreateError, SignError
+from transbank.onepay import sign
+
+from transbank import onepay
+
+class Channel(Enum):
+    WEB = "WEB"
+    MOBILE = "MOBILE"
+    APP = "APP"
+
+class Options(object):
+    def __init__(self, api_key: str, shared_secret: str):
+        self.api_key = api_key
+        self.shared_secret = shared_secret
+
+class TransbankRequestResponse(object):
+    signable_attributes = []
+
+    def get_signable_data(self, append_data = []):
+        signable_data = [getattr(self, item) for item in self.signable_attributes]
+        return signable_data + append_data
+
+class TransactionCreateRequest(TransbankRequestResponse):
+
+    signable_attributes = ['external_unique_number', 'total', 'items_quantity', 'issued_at']
+
+    def __init__(self, external_unique_number, total, items_quantity, issued_at, items,
+                 callback_url = None, channel = "WEB", app_scheme = None, options = None):
+
+        self.external_unique_number = external_unique_number
+        self.total = total
+        self.items_quantity = items_quantity
+        self.issued_at = issued_at
+        self.items = items
+        self.callback_url = callback_url
+        self.channel = channel
+        self.app_scheme = app_scheme
+        self.app_key = onepay.integration_type.value.app_key
+        self.api_key = (options or onepay).api_key
+        self.generate_ott_qr_code = True
+        self.options = options or onepay
+
+    @property
+    def signature(self):
+        return sign.build_signature_for_transaction_create_request(self, self.options.shared_secret)
```

Comment:
Now here we untangle the response object form the `sign` module (which will only become used by the `Signable` machinery, which makes sense). This method changes to:

```python
    def signature(self):
        return self.sign(self.options.shared_secret)
```

Much cleaner! :)

Reply:
Now, that's not enough. We need the `onepay.callback_url` thing you were passing from the `sign` module. I was about to tell you that this is the way to make it work:

```python
def signable_data(self):
  return super().signable_data() + [onepay.callback_url]
```

...because that's the natural way a `Signable` subclass can extend the data to be signed beyond the `signable_attributes. 

**But wait!** Why would we need to add tha callback url from the `onepay.callback_url` module? We do have a `callback_url` attribute here in this class! Why were we doing that!?

[Bonus point of the new design: by untangling the responsability it's easier to figure out weird things like the one I think I found here]

Reply:
I'd like to call the attention of @eugenioLopezRamos and @goncafa here. Perhaps @RodrigoAyala was copying the design from the PHP or Java SDK and maybe those SDKs have also a bit of a too complex thing with the signing logic?

Reply:
You would need to read all  my latest comments of the PR to get enough context, I'm afraid. Maybe @RodrigoAyala can help me getting you up to speed (if you are both together at the office tomorrow). 

But the summary is that there was a weird design on how the code handling signing logic was interacting with the objects to be signed (and verified). I fear that the Java, PHP and other SDKs might be affected by the same weird design.